core/scrollbar.js

```diff
@@ -361,23 +431,51 @@ Blockly.Scrollbar.prototype.resizeHorizontal_ = function(hostMetrics) {
  * @private
  */
 Blockly.Scrollbar.prototype.resizeVertical_ = function(hostMetrics) {
-  var outerLength = hostMetrics.viewHeight - 1;
-  if (this.pair_) {
-    // Shorten the scrollbar to make room for the corner square.
-    outerLength -= Blockly.Scrollbar.scrollbarThickness;
-  } else {
+  this.resizeViewVertical_(hostMetrics);
+  this.resizeContentVertical_(hostMetrics);
```

Comment:
At this point, is it possible to call resizeContent without calling resizeView?  It seems the barLength only changes when you call resizeView, but the bar length can change if the content size changes.  If this is what you were telling me about yesterday and this was the step towards being able to do them independently, this PR LGTM.